okhttp/src/main/java/com/squareup/okhttp/CacheControl.java

```diff
@@ -149,73 +152,93 @@ public static CacheControl parse(Headers headers) {
 
           // quoted string
-          if (pos < string.length() && string.charAt(pos) == '\"') {
+          if (pos < value.length() && value.charAt(pos) == '\"') {
             pos++; // consume '"' open quote
             int parameterStart = pos;
-            pos = HeaderParser.skipUntil(string, pos, "\"");
-            parameter = string.substring(parameterStart, pos);
+            pos = HeaderParser.skipUntil(value, pos, "\"");
+            parameter = value.substring(parameterStart, pos);
             pos++; // consume '"' close quote (if necessary)
 
             // unquoted string
           } else {
             int parameterStart = pos;
-            pos = HeaderParser.skipUntil(string, pos, ",;");
-            parameter = string.substring(parameterStart, pos).trim();
+            pos = HeaderParser.skipUntil(value, pos, ",;");
+            parameter = value.substring(parameterStart, pos).trim();
           }
         }
 
         if ("no-cache".equalsIgnoreCase(directive)) {
           noCache = true;
+          if (!isCacheControl) headerValue = null;
         } else if ("no-store".equalsIgnoreCase(directive)) {
           noStore = true;
+          if (!isCacheControl) headerValue = null;
         } else if ("max-age".equalsIgnoreCase(directive)) {
           maxAgeSeconds = HeaderParser.parseSeconds(parameter, -1);
+          if (!isCacheControl) headerValue = null;
         } else if ("s-maxage".equalsIgnoreCase(directive)) {
           sMaxAgeSeconds = HeaderParser.parseSeconds(parameter, -1);
+          if (!isCacheControl) headerValue = null;
         } else if ("public".equalsIgnoreCase(directive)) {
           isPublic = true;
+          if (!isCacheControl) headerValue = null;
         } else if ("must-revalidate".equalsIgnoreCase(directive)) {
           mustRevalidate = true;
+          if (!isCacheControl) headerValue = null;
         } else if ("max-stale".equalsIgnoreCase(directive)) {
           maxStaleSeconds = HeaderParser.parseSeconds(parameter, Integer.MAX_VALUE);
+          if (!isCacheControl) headerValue = null;
         } else if ("min-fresh".equalsIgnoreCase(directive)) {
           minFreshSeconds = HeaderParser.parseSeconds(parameter, -1);
+          if (!isCacheControl) headerValue = null;
         } else if ("only-if-cached".equalsIgnoreCase(directive)) {
           onlyIfCached = true;
+          if (!isCacheControl) headerValue = null;
         } else if ("no-transform".equalsIgnoreCase(directive)) {
           noTransform = true;
+          if (!isCacheControl) headerValue = null;
         }
       }
     }
 
     return new CacheControl(noCache, noStore, maxAgeSeconds, sMaxAgeSeconds, isPublic,
-        mustRevalidate, maxStaleSeconds, minFreshSeconds, onlyIfCached, noTransform);
+        mustRevalidate, maxStaleSeconds, minFreshSeconds, onlyIfCached, noTransform, headerValue);
   }
 
   @Override public String toString() {
+    String result = headerValue;
```

Comment:
Nice. Because you read the field exactly once, you don't need volatile for correctness.